clio-server/src/test/scala/org/broadinstitute/clio/server/dataaccess/PersistenceDAOSpec.scala

```diff
@@ -61,4 +60,39 @@ class PersistenceDAOSpec extends AsyncFlatSpec with Matchers {
       succeed
     }
   }
+
+  it should "return metadata documents from GCS in order" in {
+    val dao = new MemoryPersistenceDAO()
+    val half = 5
+    val documents = (0L until (2L * half)).map(
+      n =>
+        DocumentMock.default.copy(
+          mockKeyLong = n,
+          mockFilePath = Some(s"gs://document-mock-key-${n}")
+      )
+    )
+    val expected = documents.drop(half)
+    val result = dao
+      .initialize(index)
+      .map(
+        _ =>
+          documents
+            .map(document => dao.writeUpdate(document, index))
+      )
+      .flatMap(_ => dao.getAllSince(expected.head.upsertId, index))
+    documents.toSet.size should be(documents.size)
+    documents.map(_.upsertId).toSet.size should be(documents.size)
+    result.flatMap(_.toVector should be(expected))
+  }
+
+  it should "fail unless upsertId is found exactly once" in {
+    val upsertId = UUID.randomUUID()
+    val dao = new MemoryPersistenceDAO()
+    val result = dao
+      .initialize(index)
+      .map(_ => dao.writeUpdate(DocumentMock.default, index))
```

Comment:
This looks like it's only writing the document once, so it shouldn't throw... am I missing something?

Reply:
The write is just to initialize jimfs, which seems to need it.
The search is for some random UUID that won't be in the index.

Reply:
I couldn't get this test to work reliably so I removed it.
Feel free to help me resurrect it if you think it worthwhile.